apps/lib/http_server.c

```diff
@@ -429,11 +498,15 @@ int http_server_get_asn1_req(const ASN1_ITEM *it, ASN1_VALUE **preq,
 }
 
 /* assumes that cbio does not do an encoding that changes the output length */
-int http_server_send_asn1_resp(BIO *cbio, const char *content_type,
+int http_server_send_asn1_resp(BIO *cbio, int keep_alive,
+                               const char *content_type,
                                const ASN1_ITEM *it, const ASN1_VALUE *resp)
 {
-    int ret = BIO_printf(cbio, "HTTP/1.0 200 OK\r\nContent-type: %s\r\n"
-                         "Content-Length: %d\r\n\r\n", content_type,
+    int ret = BIO_printf(cbio, HTTP_PREFIX_VERSION"0 200 OK\r\n%s" /* HTTP/1.0 */
+                         "Content-type: %s\r\n"
+                         "Content-Length: %d\r\n\r\n",
```

Comment:
"Content-Length: %d" will result in outputting "Content-Length: 0" or "Content-Length: 1", here ?
Disclaimer: I'm not HTTP fluent ;)

Reply:
You just need to be C fluent here :)
Note that the parenthesis closes before the comparison operator:
```
ASN1_item_i2d(resp, NULL, it)) > 0
```